### PR TITLE
Add env variable and user config option to suppress packaging warnings

### PR DESCRIPTION
**Pull request by ThePrez**
### Summary
This PR adds a new env variable `QISKIT_SUPPRESS_PACKAGING_WARNINGS` and a
matching user config file option `suppress_packaging_warnings` which can be used to optionally
suppress the regular warnings about aer and/or ibmq-provider missing at import time. This strikes
a balance between telling users who potentially have a packaging issue that would otherwise go
by unnoticed and giving users who are intentionally (or unable to) install the components the ability to disable the warnings.

### Details and comments
Some platforms (in my case IBM i) cannot reliably build newer versions qiskit-aer due to new dependencies, etc. In my case, I want to pursue qiskit use with only terra (and show others), but the constant RuntimeWarning is an annoyance easily (incorrectly) perceived as an error. 

In short, I'd like a way to suppress that warning, and I think an environment variable is a cheap and easy way to do so. 

Co-authored-by: Matthew Treinish <mtreinish@kortar.org>

### CODE REVIEW

**Comment by ThePrez** [comment]
Yes, that example code using the warning filter does address my use case. I'm happy with that since it is documented (I didn't want to blindly apply a warnings filter and didn't read the doc). 

I do like both of your suggestions. `QISKIT_SUPPRESS_PACKAGING_WARNINGS` seems a good choice, and adding to the user_config certainly makes sense. Of course, I'm not sure how many people are out there building terra and aer from source. If I'm the only one who cares, we can table the idea. Or, I can clean it up and make the necessary changes.

**Comment by mtreinish** [comment]
Yeah, I think it's worthwhile even if it's not for the majority of users I think that there is a big enough set of users where this matters.

I actually hit this personally because on one of my systems I'm fighting a packaging issue with ibmq and the import doesn't ever work which always emits a warning. So having a way to filter it easily outside of the calling script would be handy.

I think that this will make the developer experience a bit more friendly. The warnings were mostly added as a stopgap against frequent bug filings because of the fragility of python packaging, especially around namespace packages. But, that was a conscious tradeoff of annoying people with more specialized use cases. So having a builtin workaround for that is a good idea.

**Comment by ajavadia** [comment]
@ThePrez is there any update on this?

**Comment by ThePrez** [comment]
Sorry. I haven't had time to circle back around and finish this up in other places. I probably won't be able to dig back into this for another couple weeks. We can let simmer, or we can merge the trivial changes in this PR and keep a work item open for the qiskit user config file and also ibmq if needed.

**Comment by mtreinish** [comment]
@ThePrez I went ahead an updated this, it's been sitting idle for several months and we'd really like to get this feature in the next release. It has come up several times that using just filter warnings isn't flexible enough for several use cases.

**Comment by mtreinish** [comment]
That's a fair point, the env variable should take precedence over the user config file. We don't have it with env vars anywhere but it does exist for the parameters (like visualization backend)

**Comment by mtreinish** [comment]
@kdk ok I updated the logic in https://github.com/Qiskit/qiskit-terra/pull/3362/commits/4f5f8eaa7c5e05e6de9d19a7ad620e32521d79ac the env var set to `N` will override the user config file setting

File: qiskit/__init__.py [hunk 1]
```diff
@@ -56,18 +56,21 @@
 try:
     from qiskit.providers.aer import Aer
 except ImportError:
-    if _config.get('suppress_packaging_warnings') or os.environ.get(
-            'QISKIT_SUPPRESS_PACKAGING_WARNINGS', '').upper() != 'Y':
+    suppress_warnings = os.environ.get('QISKIT_SUPPRESS_PACKAGING_WARNINGS', '')
+    if not _config.get('suppress_packaging_warnings') or (_config.get(
+            'suppress_packaging_warnings') and
+            suppress_warnings.upper() != 'N') or suppress_warnings.upper() != 'Y':
```
Comment (kdk) [inline qiskit/__init__.py]: There's a lint failure here. Also, I think this will always show warnings even if set in the config.

Comment (mtreinish) [reply qiskit/__init__.py]: Heh, I think pylint has gotten confused by the nested if structure:

`Wrong continued indentation (add 46 spaces).`

I guess I'll have to restructure this as nested ifs instead of one

Comment (mtreinish) [reply qiskit/__init__.py]: This should be resolved by: https://github.com/Qiskit/qiskit-terra/pull/3362/commits/a4e29d96cd926f06e4997c7780d31a6d70792b58
